private/bufpkg/bufplugin/bufpluginconfig/config.go

```diff
@@ -77,6 +87,16 @@ func newConfig(externalConfig ExternalConfig) (*Config, error) {
 			if err != nil {
 				return nil, err
 			}
+			if len(opts.overrideRemote) > 0 {
```

Comment:
Should the `bufpluginref` handle overriding the remote (maybe with a functional option passed via the `PluginReferenceForString` and co.), or did you want to scope the override to `bufpluginconfig` only?

Reply:
Added some utility functions to extract this logic out of the main loop, but I did want to keep this close to `bufpluginconfig` to start since we really only want to perform this transformation when rewriting the ExternalConfig. If we find in the future we need to have more places that transform a PluginIdentity like this, I agree that extracting it out to that package will be better.